src/Spd.Resource.Repository/SharedRepositoryFuncs.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Dynamics.CRM;
+using Spd.Resource.Repository.LicenceApplication;
+using Spd.Utilities.Dynamics;
+
+namespace Spd.Resource.Repository;
+internal static class SharedRepositoryFuncs
+{
+    public static void ProcessCategories(this DynamicsContext _context, IEnumerable<WorkerCategoryTypeEnum> categories, spd_application app)
```

Comment:
if we think these functions are common shared function. We should not use extension method. So, context will just be an input parameter. Do not make it a context extension. Context extension methods should be in Utitliy Dynamics layer.

Reply:
Fixed